### PR TITLE
PUBDEV-8414: Add support/rule coverage to RuleFit

### PR DESCRIPTION
**Pull request by koniecsveta**
…e are some discrepancies found

https://h2oai.atlassian.net/browse/PUBDEV-8414

### CODE REVIEW

File: h2o-algos/src/main/java/hex/rulefit/Rule.java [hunk 1]
```diff
@@ -15,21 +15,22 @@
     String languageRule;
     double coefficient;
     String varName;
+    double support;
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/rulefit/Rule.java:18]: does this consider weights as well?

Comment (koniecsveta) [reply h2o-algos/src/main/java/hex/rulefit/Rule.java:18]: not yet, but now it is: 
db78a07

File: h2o-genmodel/src/main/java/hex/genmodel/algos/rulefit/RuleFitMojoReader.java [hunk 2]
```diff
@@ -87,6 +87,10 @@ MojoRule readRule(String ruleId) throws IOException {
     rule._languageRule = readkv("language_rule_rule_id_" + ruleId);
     rule._coefficient = readkv("coefficient_rule_id_" + ruleId);
     rule._varName = readkv("var_name_rule_id_" + ruleId);
+    if (readkv("supprt_rule_id_" + ruleId) != null)
```
[the remark this answers is not in the document]
Comment (koniecsveta) [reply h2o-genmodel/src/main/java/hex/genmodel/algos/rulefit/RuleFitMojoReader.java]: 72150de

File: h2o-algos/src/main/java/hex/rulefit/RuleFit.java [hunk 3]
```diff
@@ -441,12 +442,15 @@ int getBestLambdaIndexCornerCase(double[] deviance, double[] lambdas) {
...
                     rule = new Rule(null, entry.getValue(), entry.getKey());
+                    // linear rule applies to all the rows
+                    rule.support = 1.0;
                 }
                 rule.setCoefficient(entry.getValue());
                 rules.add(rule);
             }
-            Comparator<Rule> ruleAbsCoefficientComparator = Comparator.comparingDouble(Rule::getAbsCoefficient).reversed();
-            rules.sort(ruleAbsCoefficientComparator);
+//            todo:  this is not needed here now, but will be when deduplication will be made before glm (mojo change needed)
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/rulefit/RuleFit.java:451]: todo?

Comment (koniecsveta) [reply h2o-algos/src/main/java/hex/rulefit/RuleFit.java:451]: will remove, this is not related to this PR

Comment (koniecsveta) [reply h2o-algos/src/main/java/hex/rulefit/RuleFit.java]: 8b1699f

**Review by maurever — COMMENTED** [review]
I did not find anything. I am just missing a test. @koniecsveta, is it possible to test this new functionality somehow? :)

**Comment by koniecsveta** [comment]
> I did not find anything. I am just missing a test. @koniecsveta, is it possible to test this new functionality somehow? :)

Will add some tests checking the final result. Basically, it is a sparseRatio of the result of SequenceProduct. SequenceProduct is tested and SparseRatio already exists.

**Review by maurever — APPROVED** [review]
LGTM.

**Review by michalkurka — APPROVED** [review]

**Pull request merged**